src/Compilers/Core/CommandLine/BuildClient.cs

```diff
@@ -93,13 +93,14 @@ internal int RunCompilation(IEnumerable<string> originalArguments, BuildPaths bu
                         return HandleResponse(buildResponse, parsedArgs, buildPaths);
                     }
                 }
-                catch
+                catch (OperationCanceledException)
                 {
-                    // It's okay, and expected, for the server compilation to fail.  In that case just fall 
-                    // back to normal compilation. 
+                    return CommonCompiler.Succeeded;
```

Comment:
On second thought -- we're passing `CancellationToken.None` above -- should we be catching this exception at all? It should be impossible.